lib/resty/etcd/v3.lua

```diff
@@ -688,43 +690,74 @@ local function request_chunk(self, method, path, opts, timeout)
 
 
     local function read_watch()
+        body = nil
 
         while(1) do
-            body, err = res.body_reader()
+            local chunk, error = res.body_reader()
+            if error then
+                return nil, error
+            end
+            if not chunk then
+                break
+            end
+
             if not body then
-                return nil, err
+                body = chunk
+            else
+                body = tab_concat({body, chunk})
```

Comment:
I think we can call tab_contat out of the while.

Reply:
Not good, loop may execute more than twice.

Reply:
this way is worse

Reply:
> Not good, loop may execute more than twice.

I mean just tab_concat once to concat all chunk

Reply:
Got it

Reply:
I thought about it carefully, and since this concat branch will only be executed in rare cases(while one chunk larger  than nginx `proxy_buffer_size`), so I think it would be more appropriate to use a string directly, which avoid creating a table with only one element in most cases.  
@membphis @spacewander @starsz Do you agree with this conclusion?

Reply:
I think just use `..` is enough.

Reply:
We can use `..` but add a comment to descript why?

Reply:
ok